--- origsrc/clamav-0.97.8/clamconf/clamconf.c	2013-04-17 10:25:08.000000000 -0500
+++ src/clamav-0.97.8/clamconf/clamconf.c	2013-05-13 10:38:51.915480500 -0500
@@ -461,7 +461,7 @@ int main(int argc, char **argv)
 	    strncpy(dbdir, pt, sizeof(dbdir));
 	    free(pt);
 	} else {
-	    strncpy(dbdir, DATADIR, sizeof(dbdir));
+	    strncpy(dbdir, CLAMAV_DATADIR, sizeof(dbdir));
 	}
 	dbdir[sizeof(dbdir) - 1] = 0;
     }
--- origsrc/clamav-0.97.8/configure.in	2013-04-17 10:25:09.000000000 -0500
+++ src/clamav-0.97.8/configure.in	2013-05-13 10:38:51.931144500 -0500
@@ -898,7 +898,7 @@ then
     fi
 fi
 
-AC_DEFINE_UNQUOTED([DATADIR],"$db_dir", [Path to virus database directory.])
+AC_DEFINE_UNQUOTED([CLAMAV_DATADIR],"$db_dir", [Path to virus database directory.])
 DBDIR="$db_dir"
 AC_SUBST([DBDIR])
 
--- origsrc/clamav-0.97.8/libclamav/readdb.c	2013-04-17 10:25:09.000000000 -0500
+++ src/clamav-0.97.8/libclamav/readdb.c	2013-05-13 10:38:51.978136500 -0500
@@ -2690,7 +2690,7 @@ int cl_load(const char *path, struct cl_
 
 const char *cl_retdbdir(void)
 {
-    return DATADIR;
+    return CLAMAV_DATADIR;
 }
 
 int cl_statinidir(const char *dirname, struct cl_stat *dbstat)
--- origsrc/clamav-0.97.8/shared/optparser.c	2013-04-17 10:25:09.000000000 -0500
+++ src/clamav-0.97.8/shared/optparser.c	2013-05-13 10:39:46.692488500 -0500
@@ -81,7 +81,7 @@ const struct clam_option __clam_options[
     { NULL, "fdpass", 0, TYPE_BOOL, MATCH_BOOL, 0, NULL, 0, OPT_CLAMDSCAN, "", "" },
     { NULL, "stream", 0, TYPE_BOOL, MATCH_BOOL, 0, NULL, 0, OPT_CLAMDSCAN, "", "" },
     { NULL, "allmatch", 'z', TYPE_BOOL, MATCH_BOOL, 0, NULL, 0, OPT_CLAMSCAN | OPT_CLAMDSCAN, "", "" },
-    { NULL, "database", 'd', TYPE_STRING, NULL, -1, DATADIR, FLAG_REQUIRED | FLAG_MULTIPLE, OPT_CLAMSCAN, "", "" }, /* merge it with DatabaseDirectory (and fix conflict with --datadir */
+    { NULL, "database", 'd', TYPE_STRING, NULL, -1, CLAMAV_DATADIR, FLAG_REQUIRED | FLAG_MULTIPLE, OPT_CLAMSCAN, "", "" }, /* merge it with DatabaseDirectory (and fix conflict with --datadir */
     { NULL, "recursive", 'r', TYPE_BOOL, MATCH_BOOL, 0, NULL, 0, OPT_CLAMSCAN, "", "" },
     { NULL, "follow-dir-symlinks", 0, TYPE_NUMBER, MATCH_NUMBER, 1, NULL, 0, OPT_CLAMSCAN, "", "" },
     { NULL, "follow-file-symlinks", 0, TYPE_NUMBER, MATCH_NUMBER, 1, NULL, 0, OPT_CLAMSCAN, "", "" },
@@ -110,8 +110,8 @@ const struct clam_option __clam_options[
     { NULL, "unpack", 'u', TYPE_STRING, NULL, -1, NULL, 0, OPT_SIGTOOL, "", "" },
     { NULL, "unpack-current", 0, TYPE_STRING, NULL, -1, NULL, 0, OPT_SIGTOOL, "", "" },
     { NULL, "info", 'i', TYPE_STRING, NULL, -1, NULL, 0, OPT_SIGTOOL, "", "" },
-    { NULL, "list-sigs", 'l', TYPE_STRING, NULL, -1, DATADIR, 0, OPT_SIGTOOL, "", "" },
-    { NULL, "find-sigs", 'f', TYPE_STRING, NULL, -1, DATADIR, FLAG_REQUIRED, OPT_SIGTOOL, "", "" },
+    { NULL, "list-sigs", 'l', TYPE_STRING, NULL, -1, CLAMAV_DATADIR, 0, OPT_SIGTOOL, "", "" },
+    { NULL, "find-sigs", 'f', TYPE_STRING, NULL, -1, CLAMAV_DATADIR, FLAG_REQUIRED, OPT_SIGTOOL, "", "" },
     { NULL, "decode-sigs", 0, TYPE_BOOL, MATCH_BOOL, 0, NULL, 0, OPT_SIGTOOL, "", "" },
     { NULL, "test-sigs", 0, TYPE_STRING, NULL, -1, NULL, 0, OPT_SIGTOOL, "", "" },
     { NULL, "vba", 0, TYPE_STRING, NULL, -1, NULL, 0, OPT_SIGTOOL, "", "" },
@@ -188,7 +188,7 @@ const struct clam_option __clam_options[
 
     { "TemporaryDirectory", "tempdir", 0, TYPE_STRING, NULL, -1, NULL, 0, OPT_CLAMD | OPT_MILTER | OPT_CLAMSCAN | OPT_SIGTOOL, "This option allows you to change the default temporary directory.", "/tmp" },
 
-    { "DatabaseDirectory", "datadir", 0, TYPE_STRING, NULL, -1, DATADIR, 0, OPT_CLAMD | OPT_FRESHCLAM | OPT_SIGTOOL, "This option allows you to change the default database directory.\nIf you enable it, please make sure it points to the same directory in\nboth clamd and freshclam.", "/var/lib/clamav" },
+    { "DatabaseDirectory", "datadir", 0, TYPE_STRING, NULL, -1, CLAMAV_DATADIR, 0, OPT_CLAMD | OPT_FRESHCLAM | OPT_SIGTOOL, "This option allows you to change the default database directory.\nIf you enable it, please make sure it points to the same directory in\nboth clamd and freshclam.", "/var/lib/clamav" },
 
     { "OfficialDatabaseOnly", "official-db-only", 0, TYPE_BOOL, MATCH_BOOL, 0, NULL, 0, OPT_CLAMD | OPT_CLAMSCAN, "Only load the official signatures published by the ClamAV project.", "no" },
 
--- origsrc/clamav-0.97.8/sigtool/sigtool.c	2013-04-17 10:25:09.000000000 -0500
+++ src/clamav-0.97.8/sigtool/sigtool.c	2013-05-13 10:38:52.025128500 -0500
@@ -1399,7 +1399,7 @@ static int listsigs(const struct optstru
 
 	mprintf_stdout = 1;
 	if(S_ISDIR(sb.st_mode)) {
-	    if(!strcmp(name, DATADIR)) {
+	    if(!strcmp(name, CLAMAV_DATADIR)) {
 		dbdir = freshdbdir();
 		ret = listdir(localdbdir ? localdbdir : dbdir, NULL);
 		free(dbdir);
