--- origsrc/clamav-0.97.8/configure.in	2013-04-17 10:25:09.000000000 -0500
+++ src/clamav-0.97.8/configure.in	2013-05-13 10:38:51.931144500 -0500
@@ -64,7 +64,7 @@ AC_CONFIG_FILES([libltdl/Makefile])
 major=`expr $LC_CURRENT - $LC_AGE`
 AC_DEFINE_UNQUOTED([LIBCLAMAV_FULLVER], "$major.$LC_AGE.$LC_REVISION",
 	  ["Full library version number"])
-AC_DEFINE_UNQUOTED([LIBCLAMAV_MAJORVER], $major,
+AC_DEFINE_UNQUOTED([LIBCLAMAV_MAJORVER], "$major",
 	  ["Major library version number"])
 AC_DEFINE([SCANBUFF], 131072, [scan buffer size])
 AC_DEFINE([FILEBUFF], 8192,   [file i/o buffer size])
