Based roughly on Debian patches

--- origsrc/clamav-0.98.6/libclamav/c++/configure.ac	2014-12-18 15:42:47.000000000 -0600
+++ src/clamav-0.98.6/libclamav/c++/configure.ac	2015-02-19 14:20:20.292380400 -0600
@@ -117,12 +117,12 @@ if test "x$llvmconfig" != "x"; then
         dnl LLVM 3.5.0 and after splits linker flags into two sets
         ldflags=`$llvmconfig --ldflags`
         syslibs=`$llvmconfig --system-libs`
-        AC_SUBST(LLVMCONFIG_LDFLAGS, ["$ldflags $syslibs"])
+        AC_SUBST(LLVMCONFIG_LDFLAGS, [])
     else
         AC_SUBST(LLVMCONFIG_LDFLAGS, [`$llvmconfig --ldflags`])
     fi
-    AC_SUBST(LLVMCONFIG_LIBS, [`$llvmconfig --libs jit nativecodegen scalaropts ipo`])
-    AC_SUBST(LLVMCONFIG_LIBFILES, [`$llvmconfig --libfiles jit nativecodegen scalaropts ipo`])
+    AC_SUBST(LLVMCONFIG_LIBS, [`$llvmconfig --libs`])
+    AC_SUBST(LLVMCONFIG_LIBFILES, [])
     AC_MSG_NOTICE([CXXFLAGS from llvm-config: $LLVMCONFIG_CXXFLAGS])
     AC_MSG_NOTICE([LDFLAGS from llvm-config: $LLVMCONFIG_LDFLAGS])
     AC_MSG_NOTICE([LIBS from llvm-config: $LLVMCONFIG_LIBS])
